libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/attributes/Attributes.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.attributes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class to create attributes for a span.
+ */
+public class Attributes {
+    private final Map<AttributeKey, Object> attributesMap;
+
+    /**
+     * Factory method.
+     * @return attributes.
+     */
+    public static Attributes create() {
+        return new Attributes();
+    }
+
+    /**
+     * Constructor.
+     */
+    private Attributes() {
+        this.attributesMap = new HashMap<>();
+    }
+
+    /**
+     * Add String attribute.
+     * @param key key
+     * @param value value
+     * @return Same instance.
+     */
+    public Attributes addAttribute(String key, String value) {
```

Comment:
Would be good to have `@Nonnull` annotation on values:

```suggestion
    public Attributes addAttribute(String key, @Nonnull String value) {
```